src/pamiq_core/time.py

```diff
@@ -0,0 +1,245 @@
+"""AMI Time Module.
+
+This module provides a custom implementation of time-related functions with time acceleration and pause/resume features.
+It wraps the standard Python time module and allows for consistent time control across the AMI system.
+
+Key features:
+- Time acceleration: Adjust the speed of time passage in the system.
+- Pause/Resume: Ability to pause and resume the flow of time in the system.
+- Thread-safe: All operations are protected by locks for use in multi-threaded environments.
+- Compatible API: Provides familiar time functions like sleep, time, perf_counter, and monotonic.
+- Type annotations: Utilizes Python 3.10+ type annotations for improved type checking and IDE support.
+- Original time functions: Provides access to the original time functions with 'fixed_' prefix.
+
+Usage:
+    from ami import time
+
+    # Get current time (affected by time scale and pause)
+    current_time = time.time()
+
+    # Get current time (not affected by time scale or pause)
+    fixed_current_time = time.fixed_time()
+
+    # Sleep for 1 second (affected by time scale and pause)
+    time.sleep(1)
+
+    # Sleep for 1 second (not affected by time scale or pause)
+    time.fixed_sleep(1)
+
+    # Set time scale (e.g., 2x speed)
+    time.set_time_scale(2.0)
+
+    # Get current time scale
+    current_scale = time.get_time_scale()
+
+    # Pause time
+    time.pause()
+
+    # Resume time
+    time.resume()
+
+Note: This module is designed for use within the AMI system and may not be suitable for general-purpose time management.
+"""
+
+from __future__ import annotations
+
+import time as _original_time
+from functools import wraps
+from threading import RLock
+from typing import Callable, Concatenate, ParamSpec, TypedDict, TypeVar
+
+T = TypeVar("T")
+P = ParamSpec("P")
```

Comment:
型パラメータはすでに参照されていないので、削除する必要があります。Python3.12の型構文からのアップデートです。
```suggestion
from typing import Callable, Concatenate, TypedDict

```